shells/android/java/arcs/crdt/parcelables/ParcelableCrdtSet.kt

```diff
@@ -0,0 +1,212 @@
+                    val clock = parcel.readTypedObject(ParcelableVersionMap.CREATOR)!!.actual
+                    val actor = requireNotNull(parcel.readString())
+                    val added = requireNotNull(parcel.readReferencable())
+                    return Add(CrdtSet.Operation.Add(clock, actor, added))
+                }
+
+                override fun newArray(size: Int): Array<Add?> = arrayOfNulls(size)
+            }
+        }
+
+        /** Parcelable variant of [CrdtSet.Operation.Remove]. */
+        data class Remove(
+            override val actual: CrdtSet.Operation.Remove<Referencable>
+        ) : Operation(OpType.Remove) {
+            override fun describeContents(): Int = 0
+
+            override fun writeToParcel(parcel: Parcel, flags: Int) {
+                super.writeToParcel(parcel, flags)
+                parcel.writeTypedObject(ParcelableVersionMap(actual.clock), flags)
+                parcel.writeString(actual.actor)
+                parcel.writeTypedObject(ParcelableReferencable(actual.removed), flags)
+            }
+
+            companion object CREATOR : Parcelable.Creator<Remove> {
+                override fun createFromParcel(parcel: Parcel): Remove {
+                    val clock = parcel.readTypedObject(ParcelableVersionMap.CREATOR)!!.actual
+                    val actor = requireNotNull(parcel.readString())
+                    val removed = requireNotNull(parcel.readReferencable())
+                    return Remove(CrdtSet.Operation.Remove(clock, actor, removed))
+                }
+
+                override fun newArray(size: Int): Array<Remove?> = arrayOfNulls(size)
+            }
+        }
+
+        /** Parcelable variant of [CrdtSet.Operation.FastForward]. */
+        data class FastForward(
+            override val actual: CrdtSet.Operation.FastForward<Referencable>
+        ) : Operation(OpType.FastForward) {
+            override fun describeContents(): Int = 0
+
+            override fun writeToParcel(parcel: Parcel, flags: Int) {
+                super.writeToParcel(parcel, flags)
+                parcel.writeTypedObject(ParcelableVersionMap(actual.oldClock), flags)
+                parcel.writeTypedObject(ParcelableVersionMap(actual.newClock), flags)
+
+                parcel.writeInt(actual.added.size)
+                actual.added.forEach {
+                    parcel.writeTypedObject(DataValue(it), flags)
+                }
+
+                parcel.writeInt(actual.removed.size)
+                actual.removed.forEach {
+                    parcel.writeTypedObject(ParcelableReferencable(it), flags)
+                }
+            }
+
+            companion object CREATOR : Parcelable.Creator<FastForward> {
+                override fun createFromParcel(parcel: Parcel): FastForward {
+                    val oldClock = parcel.readTypedObject(ParcelableVersionMap.CREATOR)!!.actual
+                    val newClock = parcel.readTypedObject(ParcelableVersionMap.CREATOR)!!.actual
```

Comment:
Same, I'll omit the suggestion from here on out, but `requireNotNull` is preferred to `!!` in general

Reply:
Done